src/Renderer/Camera.js

```diff
@@ -100,4 +101,26 @@ Camera.prototype.box3SizeOnScreen = function box3SizeOnScreen(box3, matrixWorld)
     return tmp.box3;
 };
 
+ /**
+ * Test for collision between camera and a geometry layer (DTM/DSM) to adjust camera position
+ * It could be modified later to handle an array of geometry layers
+ * @param {view} view where we test the collision between geometry layers and the camera
+ * @param {elevationLayer} elevationLayer (DTM/DSM) used to test the collision with the camera. Could be another geometry layer
+ * @param {minDistanceCollision} minDistanceCollision the minimum distance allowed between the camera and the surface
+ */
+Camera.prototype.adjustAltitudeToAvoidCollisionWithLayer = function adjustAltitudeToAvoidCollisionWithLayer(view, elevationLayer, minDistanceCollision) {
+    const camLocation = new Coordinates(view.referenceCrs, view.camera.camera3D.position).as('EPSG:4326');
+    if (elevationLayer !== undefined) {
+        const elevationUnderCamera = DEMUtils.getElevationValueAt(elevationLayer, camLocation);
+        if (elevationUnderCamera != undefined) {
+            const difElevation = camLocation.altitude() - (elevationUnderCamera.z + minDistanceCollision);
+            // We move the camera to avoid collisions if too close to terrain
+            if (difElevation < 0) {
+                camLocation.setAltitude(elevationUnderCamera.z + minDistanceCollision);
+                view.camera.camera3D.position.copy(camLocation.as(view.referenceCrs).xyz());
```

Comment:
cf: https://github.com/iTowns/itowns/pull/465#discussion_r138268173 
"I'm not sure all controls instances support external changes of camera's position."

Did you check?

Reply:
You probably should use `view.notifyChange(...)` here since you modify the camera position.

Reply:
The three different controls we use support it.
But yes I thought in the future about having a function in camera that just compute a new position if collision so the controller itself can modify the camera position using this value.

Reply:
Ok, let's keep it this way if it's working fine with the available controls.

Reply:
(We don't need view.notifyChange as this function is called on a control update which then notify the change)

Reply:
> We don't need view.notifyChange [...]

You need to call `view.notifyChange` because you're modifying the camera's position.
The fact that the controls update method may call `view.notifyChange` too is irrelevant.

Reply:
yes i did it in fact it's rebased

Reply:
yes i did it in fact it's rebased